demos/projects/ESPRESSIF/az-ca-recovery/components/sample-azure-iot-ca-recovery/azure_trust_bundle_storage_esp.c

```diff
@@ -0,0 +1,185 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#include <string.h>
+
+#include "azure_trust_bundle_storage.h"
+
+#include "esp_system.h"
+#include "esp_log.h"
+#include "nvs_flash.h"
+#include "nvs.h"
+
+static const char * TAG = "trust-bundle-storage";
+
+#define CA_CERT_NAMESPACE                  "root-ca-cert"
+#define AZURE_TRUST_BUNDLE_NAME            "az-tb"
+#define AZURE_TRUST_BUNDLE_VERSION_NAME    "az-tb-ver"
+
+AzureIoTResult_t AzureIoTCAStorage_ReadTrustBundle( const uint8_t * pucTrustBundle,
+                                                    uint32_t ulTrustBundleLength,
+                                                    uint32_t * pulOutTrustBundleLength,
+                                                    const uint8_t * pucTrustBundleVersion,
+                                                    uint32_t ulTrustBundleVersionLength,
+                                                    uint32_t * pulOutTrustBundleVersionLength )
+{
+    nvs_handle_t xNVSHandle;
+    esp_err_t err;
+    size_t ulTrustBundleReadSize = 0; /* value will default to 0, if not set yet in NVS */
+    size_t ulTrustBundleVersionReadSize = 0;
+
+    err = nvs_open( CA_CERT_NAMESPACE, NVS_READWRITE, &xNVSHandle );
+
+    if( err != ESP_OK )
+    {
+        ESP_LOGE( TAG, "Error (%s) opening NVS!\n", esp_err_to_name( err ) );
+        return err;
+    }
+
+    err = nvs_get_blob( xNVSHandle, AZURE_TRUST_BUNDLE_VERSION_NAME, NULL, &ulTrustBundleVersionReadSize );
+
+    if( err != ESP_OK )
+    {
+        ESP_LOGE( TAG, "Error (%s) getting AZURE_TRUST_BUNDLE_VERSION_NAME from NVS!\n", esp_err_to_name( err ) );
+        nvs_close( xNVSHandle );
+        return err;
```

Comment:
Will these errors be `AzureIoTResult_t` errors, or do they need to be converted?